### PR TITLE
CIRCSTORE-91: Patron notice policy - Get All

### PR DESCRIPTION
**Pull request by opochepam**
Circulation storage - get patron notice policy collection.

Jira link: [CIRCSTORE-91](https://issues.folio.org/browse/CIRCSTORE-91)

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java [hunk 1]
```diff
@@ -23,9 +28,55 @@
...
+    Context vertxContext) {
+
+    vertxContext.runOnContext(v -> {
+      try {
+        String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+        PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+
+        String[] fieldList = {"*"};
+
+        CQL2PgJSON cql2pgJson = new CQL2PgJSON("patron_notice_policy.jsonb");
```
Comment (dmytrotur) [inline src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java]: Seems like there is constant value defined for table name

Comment (opochepam) [reply src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java]: replaced by `PATRON_NOTICE_POLICY_TABLE + ".jsonb"`

File: ramls/patron-notice-policies.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Collection of patron notice policies",
+  "type": "object",
+  "properties": {
+    "patronNoticePolicies": {
+      "description": "List of items",
```
Comment (julianladisch) [inline ramls/patron-notice-policies.json]: List of patron notice policies

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java [hunk 3]
```diff
@@ -101,6 +115,27 @@ public void cannotUpdateNonexistentPatronNoticePolicy() throws InterruptedExcept
...
+
+    CompletableFuture<JsonResponse> getAllCompleted = new CompletableFuture<>();
+
+    client.get(patronNoticePoliciesStorageUrl(), StorageTestSuite.TENANT_ID, ResponseHandler.json(getAllCompleted));
+
+    JsonResponse response = getAllCompleted.get(5, TimeUnit.SECONDS);
+
+    MatcherAssert.assertThat("Failed to get all patron notice policies", response.getStatusCode(), CoreMatchers.is(200));
+    MatcherAssert.assertThat(response.getJson().getJsonArray("patronNoticePolicies").size(), CoreMatchers.is(2));
+    MatcherAssert.assertThat(response.getJson().getInteger("totalRecords"), CoreMatchers.is(2));
```
Comment (julianladisch) [inline src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: There needs to be a check that the response actually contains the names "firstPolicy" and "secondPolicy"

File: ramls/patron-notice-policies.json (hunk 2 above)
Comment (opochepam) [reply ramls/patron-notice-policies.json]: changed to "List of patron notice policies"

File: ramls/patron-notice-policy.raml [hunk 4]
```diff
@@ -10,11 +10,33 @@ documentation:
...
+  pageable: !include raml-util/traits/pageable.raml
+  searchable: !include raml-util/traits/searchable.raml
+
 /patron-notice-policy-storage:
   /patron-notice-policies:
     displayName: Patron Notice Policies
+    get:
+      is: [
+        pageable,
+        searchable: {description: "searchable using CQL", example: "id=\"cf23adf0-61ba-4887-bf82-956c4aae2260\""}
```
[the remark this answers is not in the document]
Comment (opochepam) [reply ramls/patron-notice-policy.raml]: changed to `"name=\"undergrad*\""`

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java (hunk 3 above)
Comment (opochepam) [reply src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: added name check

Comment (julianladisch) [reply src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Thanks.
This line is duplicate: `MatcherAssert.assertThat(policies.size(), CoreMatchers.is(2));`
There is no "sortBy" so the database may return the policies in any order. You can use a matcher that matches both names but in any order: http://hamcrest.org/JavaHamcrest/javadoc/2.0.0.0/org/hamcrest/collection/IsArrayContainingInAnyOrder.html

Comment (opochepam) [reply src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Thank you. Rewrited the code using `arrayContainingInAnyOrder`.

**Pull request merged**